### ISSUE TITLE
QUESTION: Native Bindings

### ISSUE THREAD

**Issue by hossam14102010** [sipsorcery-org/sipsorcery#1172]
Does this library use WebRTC native bindings, and if it does not do so, why not?

**Comment by DavidMartynWood** [comment]
Seems like rather an obnoxious question. This is an open source project so:
1. You could look at the code
2. You could suggest a reason as to why it would be helpful to do whatever it is you think it should be doing
3. You could submit a PR
And while I don't know what "native webrtc bindings" are. I note that this project is largely cross platform and using "native" libraries has extra considerationa.
I am not affiliated with this project, just thought you sounded rude so wanted to chip in.

**Comment by ha-ves** [comment]
### Quoting from another WebRTC .NET implementation library:
> ...current way of incorporating the WebRTC libraries into your project is a total nightmare. The build system (aka Ninja http://www.webrtc.org/native-code/development) is extremely complicated. You are going to be doing everything from the command line, it take ages to even download the source code (~10GB), try it and cry... ;D

If you found out which library it is, look at when was the last time it was updated...
That's what happens if you try to bind/wrap it.

I presume you had a few reads about WebRTC...
If not, I encourage you to look up the history of WebRTC, here are some summaries:

## WebRTC has two parts:
- IETF (and by extension the W3C)'s Specifications, namely known as RFCs.
- Implementations, yes s-es,

Google did not make it to a library, it is heavily dependent on Chromium, and it is foremost a web browser feature.

Microsoft does implement WebRTC for Edge. And so does Mozilla and other web browsers, etc., etc.

They even had a C# .NET implementation with [Microsoft.MixedReality.WebRTC](https://github.com/microsoft/MixedReality-WebRTC) but it got archived in 2022 which seems too coincidental with their leaving the involvement since 2019.

But that one got picked up by the Unity3D Game Engine community and made their forks. (I haven't checked more on this if they made it work for other platforms)

I think they moved it to the Azure Real-Time Communication but that is not strictly WebRTC...

## As for this library...
<details>
<summary>I'll leave it for the maker to describe why they made it.</summary>
 I'm not trying to conspire or anything here, but this repository was kinda heading in the same direction... unsupported and forgotten... don't ban me.

---

</details>

What I saw is that it functions WebRTC, and is platform-agnostic.

# To answer your question,
In short, binding the "native WebRTC" would need you to at least build a "web browser."

---

<details>
<summary>P.S</summary>
I saw a response in email from another person, it seems like it has gone, well I dont know about that one.
</details>
